lib/services/post.ts

```diff
@@ -0,0 +1,158 @@
+import { Post, PostStatus, User } from "@prisma/client";
+import { prisma } from "./db";
+import { deleteImage, uploadImage } from "./image";
+
+export type PostWithUser = Post & {
+  user: User;
+};
+
+export type CreatablePost = {
+  price: number;
+  description: string;
+  image?: Buffer;
+  bookId: string;
+  userId: string;
+};
+
+export type UpdatablePost = {
+  price?: number;
+  description?: string;
+  image?: Buffer;
+  status?: PostStatus;
+};
+
+type UpdatablePostResolved = {
+  price?: number;
+  description?: string;
+  imageUrl?: string;
+  status?: PostStatus;
+};
+
+/**
+ * Returns the post with the given id.
+ *
+ * @param id the id of the post
+ * @param includeUser whether or not the user associated with the post should be returned in the data
+ * @returns the post with the given id, or null if the post with the given id cannot be found
+ */
+export async function getPost(
+  id: string,
+  includeUser: boolean
+): Promise<Post | PostWithUser | null> {
+  return prisma.post.findUnique({
+    where: {
+      id: id,
+    },
+    include: {
+      user: includeUser,
+    },
+  });
+}
+
+/**
+ * Creates a post from the given information, then returns the post.
+ *
+ * Uploads the given image.
+ *
+ * @param creatablePost the information used to create the posts
+ * @returns the created post
+ */
+export async function createPost(creatablePost: CreatablePost): Promise<Post> {
+  let imageUrl: string | undefined = undefined;
+  if (!!creatablePost.image) {
```

Comment:
Shouldn't need to convert to a boolean here. `!!` is essentially a noop in this case.

Reply:
Okay. I just wasn't sure if we wanted to have explicit casting to show intent. I'll remove the casting